lib/PuppeteerSharp/Page.cs

```diff
@@ -1865,7 +1866,7 @@ private async Task<object> ExecuteBinding(BindingCalledResponse e)
                 if (taskResult.GetType().IsGenericType)
                 {
                     // the task is already awaited and therefore the call to property Result will not deadlock
-                    result = ((dynamic)taskResult).Result;
+                    result = taskResult.GetType().GetProperty(taskResultPropertyName).GetValue(taskResult);
```

Comment:
the old way was weird but this also feels a bit odd, is there a clean way to do this?

Reply:
I open to new ideas 😂

Reply:
Created an issue for this https://github.com/kblok/puppeteer-sharp/issues/824